### ISSUE TITLE
Embed window controls ( - + x ) / traffic lights in the app when running on Linux

### ISSUE THREAD

**Issue by rathboma** [electron/electron#23665]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description
Breaking out from #11907 currently there is no way on Linux to embed the native window controls (traffic lights) inside the Electron app.

This currently works great on MacOS, and allows us to implement our own good-looking (if not entirely native) titlebar UI, or omit it entirely.

Mac screenshot:
![Screen Shot 2020-05-19 at 10 41 00 AM](https://user-images.githubusercontent.com/279769/82347473-442eae00-99bd-11ea-9600-3b6f9731d38a.png)

### Proposed Solution
- A config setting to allow Traffic lights ( - + x ) to be embedded.
- Some sort of property to tell the app where on the app the traffic lights are (left, middle, right), and the horizontal size

### Alternatives Considered
See #11907 for a list of alternatives

**Comment by Exalm** [comment]
This is kinda difficult, but doable: you would have to look at GtkHeaderbar's window button implementation and basically copy that.

In GTK4 there's a widget for this specifically, alled `GtkWindowControls`, that you could overlay about your view though.

Also don't forget the controls can be on either side of the window, incl. both at once.

**Comment by rathboma** [comment]
If this is easily doable using GTK4 I think that's a great path forward. For any DE supporting GTK4 it could allow this feature.

**Comment by Exalm** [comment]
it's not DE that should support GTK4, it's Electron. A window is a window regardless of what toolkit it uses to draw it. However, GTK4 is not released yet. :p

**Comment by rathboma** [comment]
Yep yep I know we're only on 3.38 :-), but it's nice to have a forward looking idea if possible!

**Comment by Exalm** [comment]
3.38? Umm, 3.24... But ok.

**Comment by Exalm** [comment]
Just forgot, another thing you will need in this case is to mask away window corners (the only reason headerbars have them is because they are not in the main window and so there's no content there, and bottom corners are impossible). I hope to get that into GTK4, but if you go with GTK3, there's a widget for that in libhandy (HdyWindow)
